app/models/user.rb

```diff
@@ -29,6 +42,20 @@ def self.from_omniauth(auth)
   end
 
 private
+  def check_email_format
+    return if errors.key?(:email)
+    validates_format_of :email, with: /\A([a-z0-9_\.-]+\@[\da-z\.-]+\.[a-z\.]{2,6})\z/, message: "Address Invalid Format"
+  end
+
+  def check_phone_format
+    return if errors.key?(:phone)
+    validates_format_of :phone, with: /\d{3}[-\s]?\d{3}[-\s]?\d{4}/, message: "Number Invalid Format"
```

Comment:
This is cool. Doubt it's worth the effort, but I always appreciated those form inputs that either limit each input field to 3 or 4 characters, or automatically insert the '-' and '( )' into phone numbers.